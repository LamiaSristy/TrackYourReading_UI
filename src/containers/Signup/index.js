/* eslint-disable camelcase */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import './Signup.css';
import { createUser } from '../../actions/user';

class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      password_confirmation: '',
      errors: '',
    };
  }

  handleChangeName = e => {
    this.setState({
      username: e.target.value,
    });
  }

  handleChangePassword = e => {
    this.setState({
      password: e.target.value,
    });
  }

  handleChangePasswordConfirm = e => {
    this.setState({
      password_confirmation: e.target.value,
    });
  }

   handleSubmit= async e => {
     e.preventDefault();
     const {
       username, password, password_confirmation,
     } = this.state;
     const { createUser } = this.props;

     const response = await createUser({ username, password, password_confirmation });
     if (response && response.status === 200) {
       const { history } = this.props;
       history.push('/main');
     } else {
       const { error } = this.props;
       this.setState({
         errors: error,
       });
     }
   }

   handleErrors = () => {
     const { errors } = this.state;
     setTimeout(() => this.setState({ errors: '' }), 3000);
     return (
       <ul>
         {errors.map(error => <li key={error}>{error}</li>)}
       </ul>
     );
   }

   render() {
     const {
       username, errors, password, password_confirmation,
     } = this.state;
     return (
       <section className="signup">
         <div className="errors-div">
           {errors ? this.handleErrors() : null}
         </div>
         <h2>Sign Up</h2>
         <form onSubmit={this.handleSubmit}>
           <input
             placeholder="Username"
             type="text"
             name="username"
             value={username}
             onChange={this.handleChangeName}
             required
           />
           <input
             placeholder="Password"
             type="password"
             name="password"
             value={password}
             onChange={this.handleChangePassword}
             required
           />
           <input
             placeholder="Password confirmation"
             type="password"
             name="password_confirmation"
             value={password_confirmation}
             onChange={this.handleChangePasswordConfirm}
             required
           />
           <button placeholder="submit" type="submit">
             Sign In
           </button>
         </form>
       </section>
     );
   }
}

const mapStateToProps = state => ({
  user: state.user,
  isLogin: state.user.isLogin,
  error: state.user.error,
});
const mapDispatchToProps = dispatch => ({
  createUser: data => dispatch(createUser(data)),
});

Signup.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
  createUser: PropTypes.func.isRequired,
  error: PropTypes.instanceOf(Array),

};

Signup.defaultProps = {
  error: [],
  history: {},
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Signup));
