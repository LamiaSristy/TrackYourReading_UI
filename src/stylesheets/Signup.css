@import "https://fonts.googleapis.com/css2?family=Raleway:wght@700&display=swap";
@import "https://fonts.googleapis.com/css2?family=Roboto&display=swap";

h2 {
  color: #721c24;
}

.signup {
  margin: 0 auto;
}

.signup h2 {
  text-align: center;
  font-family: 'Raleway', sans-serif;
}

.signup form {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Roboto', sans-serif;
}

.signup form input {
  height: 28px;
  width: 238px;
  border: 1px dotted #721c24;
  margin: 7px 0;
  border-radius: 5px;
  padding-left: 10px;
}

.signup form input::placeholder {
  padding: 0 10px;
}

.signup form button {
  height: 40px;
  width: 250px;
  border: none;
  background-color: rgb(69, 114, 69);
  padding: 0 20px;
  border-radius: 5px;
  color: white;
  font-weight: bold;
  cursor: pointer;
  margin: 20px 0;
}
