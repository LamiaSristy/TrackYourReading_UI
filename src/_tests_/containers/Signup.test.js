import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Signup from '../../containers/Signup';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Signup />
  </Provider>,
);

describe('Signup Container', () => {
  let container;
  beforeEach(() => {
    container = setup();
  });

  test('renders the signup class in the page', () => {
    const page = container.find('.signup');
    expect(page).toMatchSnapshot();
  });

  test('renders the text of the errors-div class', () => {
    const text = container.find('.errors-div');
    expect(text).toMatchSnapshot();
  });
});
