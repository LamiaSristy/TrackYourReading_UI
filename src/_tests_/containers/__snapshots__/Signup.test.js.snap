// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Container renders the signup class in the page 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the add-book class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the book-description class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the book-info class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the book-name class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the books class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the buttons class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the errors-div class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the one-book class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the text of the tracking class 1`] = `ShallowWrapper {}`;

exports[`Signup Container renders the the page 1`] = `ShallowWrapper {}`;
